src/reactor/cache.js

```diff
@@ -0,0 +1,201 @@
+import { List, Map, OrderedSet, Record } from 'immutable'
+
+export const CacheEntry = Record({
+  value: null,
+  storeStates: Map(),
+  dispatchId: null,
+})
+
+/*******************************************************************************
+ * interface PersistentCache {
+ *    has(item)
```

Comment:
Why are `lookup`, `has`, and `hit` all necessary? Is a `get` method insufficient?